Cesium3DTilesReader/test/TestTilesetReader.cpp

```diff
@@ -310,3 +310,85 @@ TEST_CASE("Reads custom extension") {
   REQUIRE(zeroExtensions.empty());
 }
+
+TEST_CASE("Reads tileset JSON with unknown properties") {
+  using namespace std::string_literals;
+
+  std::filesystem::path tilesetFile = Cesium3DTilesReader_TEST_DATA_DIR;
+  tilesetFile /= "tileset-with-unsupported-properties.json";
+  std::vector<std::byte> data = readFile(tilesetFile);
+  Cesium3DTilesReader::TilesetReader reader;
+  Cesium3DTilesReader::TilesetReaderResult result = reader.readTileset(data);
+  CHECK(result.errors.empty());
+  CHECK(result.warnings.empty());
+  REQUIRE(result.tileset);
+
+  const CesiumUtility::JsonValue::Object& unknownProperties =
+      result.tileset->asset.unknownProperties;
+
+  auto itString = unknownProperties.find("someString");
+  REQUIRE(itString != unknownProperties.end());
+  REQUIRE(itString->second.isString());
+  REQUIRE(itString->second.getString() == "A");
+
+  auto itDouble = unknownProperties.find("someDouble");
+  REQUIRE(itDouble != unknownProperties.end());
+  REQUIRE(itDouble->second.isDouble());
+  REQUIRE(itDouble->second.getDouble() == 2.1);
+
+  auto itInt = unknownProperties.find("someInt");
+  REQUIRE(itInt != unknownProperties.end());
+  REQUIRE(itInt->second.isUint64());
+  REQUIRE(itInt->second.getUint64() == 5);
+
+  auto itSignedInt = unknownProperties.find("someSignedInt");
+  REQUIRE(itSignedInt != unknownProperties.end());
+  REQUIRE(itSignedInt->second.isInt64());
+  REQUIRE(itSignedInt->second.getInt64() == -5);
+
+  auto itBool = unknownProperties.find("someBool");
+  REQUIRE(itBool != unknownProperties.end());
+  REQUIRE(itBool->second.isBool());
+  REQUIRE(itBool->second.getBool() == true);
+
+  auto itArray = unknownProperties.find("someArray");
+  REQUIRE(itArray != unknownProperties.end());
+  REQUIRE(itArray->second.isArray());
+  const CesiumUtility::JsonValue::Array& array = itArray->second.getArray();
+  REQUIRE(array.size() == 1);
+  REQUIRE(array[0].isString());
+  REQUIRE(array[0].getString() == "hi");
+
+  auto itObject = unknownProperties.find("someObject");
+  REQUIRE(itObject != unknownProperties.end());
+  REQUIRE(itObject->second.isObject());
+  const CesiumUtility::JsonValue::Object& o = itObject->second.getObject();
+  REQUIRE(o.size() == 1);
+  auto itObjectValue = o.find("value");
+  REQUIRE(itObjectValue != o.end());
+  REQUIRE(itObjectValue->second.isString());
+  REQUIRE(itObjectValue->second.getString() == "test");
+
+  auto itNull = unknownProperties.find("someNull");
+  REQUIRE(itNull != unknownProperties.end());
+  REQUIRE(itNull->second.isNull());
+}
+
+TEST_CASE("Reads tileset JSON with unknown properties and ignores them when "
+          "requested") {
+  using namespace std::string_literals;
+
+  std::filesystem::path tilesetFile = Cesium3DTilesReader_TEST_DATA_DIR;
+  tilesetFile /= "tileset-with-unsupported-properties.json";
+  std::vector<std::byte> data = readFile(tilesetFile);
+  Cesium3DTilesReader::TilesetReader reader;
+  reader.getOptions().setCaptureUnknownProperties(false);
```

Comment:
For my curiosity, is there a compelling reason to have the option to turn on / off the capture? My guess is it saves performance / memory having it off.

Reply:
Yeah that's about it, though the effect should be small (barring weird edge cases like a glTF will a huge number of unknown properties for some reason). I probably could have skipped this.